Examples/Algorithms/TrackFinding/src/SpacePointMaker.cpp

```diff
@@ -98,9 +98,9 @@ ActsExamples::SpacePointMaker::SpacePointMaker(Config cfg,
   spBuilderConfig.trackingGeometry = m_cfg.trackingGeometry;
 
   m_slSurfaceAccessor.emplace(
-      IndexSourceLink::SurfaceAccessor{*m_cfg.trackingGeometry});
+      IndexSourceLinkSurfaceAccessor{m_cfg.trackingGeometry.get()});
```

Comment:
could you typdef `SurfaceAccessor` inside `IndexSourceLink`? e.g.
```
using SurfaceAccessor = IndexSourceLinkSurfaceAccessor;
```
that would reduce the diff a bit

same for the test one

Reply:
Won't it look kinda awkward if you'll try to use the `Experimental` with the implementations? Like, you already have a typedef inside the struct, but you have to go out of your way and plug in the external implementation.

Reply:
What I would try to avoid, if possible, is to change existing code to make `Experimental` stuff fit. If we typedef the original one the existing Gen1 code will not change. For the Experimental one we can just use a none typedef'd one

Reply:
Done